core/src/main/java/org/opensearch/sql/executor/execution/PaginatedPlan.java

```diff
@@ -38,8 +39,9 @@ public void execute() {
   }
 
   @Override
+  // TODO why can't use listener given in the constructor?
   public void explain(ResponseListener<ExecutionEngine.ExplainResponse> listener) {
-    listener.onFailure(new UnsupportedOperationException(
+    listener.onFailure(new NotImplementedException(
```

Comment:
Why change from `UnsupportedOperationException` to `NotImplementedException` here but from `NotImeplementedException` to `UnsuportedException` `ContinuePaginatedPlane.java`?

Reply:
Just to comply with nature of these exceptions. Explain for `PaginatedPlan` will be implemented soon (so throw `NotImplementedException`), but explain in `ContinuePaginatedPlan` won't be implemented ever (as I understand) - so `UnsupportedOperationException` is thrown.